shared-libs/outbound/src/outbound.js

```diff
@@ -177,15 +179,39 @@ const sendPayload = (payload, config) => {
   });
 };
 
-const updateInfo = (recordInfo, configName) => {
+// Never change this hashing algorithm or how we stringify, otherwise you will invalidate all existing hashes
+const hash = payload => crypto.createHash('sha256').update(JSON.stringify(payload)).digest('hex');
```

Comment:
There is a potential bug here. Firstly `JSON.stringify` is not deterministic so you may get properties in a different order (if you upgrade Node for example). Secondly I'm not sure our payload generation will be deterministic either. 

I'm not sure how to solve this without deep sorting the object properties. Consider using this library: https://www.npmjs.com/package/object-hash

Reply:
Awesome catch. I wasn't aware that stringify wasn't deterministic, but you're right it definitely is at least based on how the object gets constructed, which if anything we could accidentally change pretty easily.

So what are your thoughts on:
```
const hash = payload => require('object-hash')(JSON.parse(JSON.stringify(payload)));
```

Notes:
 - `stringify` then `parse` to make sure it's the contents that would be sent over the wire that we're hashing, not stuff that wouldn't make it.
 - all the default options seem good to me, and I presume they wouldn't be so bold as to change defaults without also bumping a major

The other option is that we just store the entire JSON object with no hashing at all, and use a deep equality check instead of checking the hash. The upside is that this is *also* nice from a logging / debugging / analysis perspective, as you can see and debug what was sent. The downsides are the obvious ones around doc size, esp in situations where we are keeping docs maybe aggressively in sync. We could purge them after they are needed (ie if you sent A,B,A you don't need to keep the first A) but then you arguably lose some of the pros because you can't see the whole history. Though, I already [raised a ticket for this problem in general](#6464) so maybe if we trust we'll solve that issue then the size doesn't matter?

Reply:
@dianabarsan interested in your thoughts here as well, if you have any (no pressure to!)

Reply:
I would not store the whole json unless we limit stored history. It brings back memories about audit-db saves crashing instances. 
`object-hash` might be ok, but then we're locked into not updating it, unless we want to risk getting different hashes cause of changes in their code - it looks like they have their own algorithm that generates the hashable string.

Reply:
Yeah I think it's important to lock the version longterm, though without comments in the package.json it's going to be more finicky to do.

Reply:
It'd be easy enough to write a test to parse the package.json (or package-lock.json) and verify the number.

Or instead, do you think it'd be possible to write enough integration tests to be confident that if the underlying implementation did change in any substantive way we would find out?